controller/scheduler-v2/scheduler.go

```diff
@@ -446,11 +453,9 @@ func (s *Scheduler) HandleFormationChange(ef *ct.ExpandedFormation) {
 	}()
 
 	log := logger.New("fn", "HandleFormationChange")
-	if ef.App != nil {
-		log = log.New("app.id", ef.App.ID)
-	}
-	if ef.Release != nil {
-		log = log.New("release.id", ef.Release.ID)
+	if ef == nil || ef.App == nil || ef.Release == nil {
+		log.Error("invalid formation")
+		return
 	}
```

Comment:
This should no longer be necessary as the sentinel is [explicitly ignored in streamFormationEvents](https://github.com/flynn/flynn/blob/scheduler-v2-review/controller/scheduler-v2/scheduler.go#L182-L185), so if we are getting a nil formation it's a bug.